skeletons/chrome/content-script.js

```diff
@@ -87,22 +87,37 @@
   }
 
   /**
-   * Gather the iframes running in the ClientApp
-   */
-  var iframes = document.getElementsByTagName('iframe');
-  var urls = [];
-  for (var i = 0, l = iframes.length; i < l; i ++) {
-    urls.push(iframes[i].src);
-  }
-
-  /**
    * Send the iframes to EmberInspector so that it can run
    * EmberDebug in the context of the iframe.
    */
-  //FIX ME
-  setTimeout(function() {
-    chrome.extension.sendMessage({type: 'iframes', urls: urls});
-  }, 500);
 
+  function sendFrameToInspector(url) {
+    chrome.extension.sendMessage({type: 'iframes', urls: [url]});
+  }
+
+  // create an observer instance for body tag, so if any iframes get pushed
+  // to DOM we attach onLoad event listener on it and when it has finally loaded
+  // we send it's URL to Inspector, otherwise we might send URL but Ember app in the
+  // frame is still loading and thus will not be detected
+  var forEach = Array.prototype.forEach;
+  var bodyObserver = new MutationObserver(function(mutations) {
+    mutations.forEach(function(mutation) {
+      forEach.call(mutation.addedNodes, function(node){
```

Comment:
It _may_ be faster to do a `MutationObserver` + `getElementsByTagName` than iteration over all `addedNodes`.